### ISSUE TITLE
Update data_model.rst file with latest changes

### ISSUE THREAD

**Issue by RDmitchell** [SEED-platform/seed#645]
from @rebrownlbnl 
This passage is no longer correct in the data_model.rst file:
* Here BS1 entered the system and was matched with BS0. When a match occurs, a new BuildingSnapshot is created, BS2, with the fields from the primary BuildingSnapshot, BS0, and the secondary BuildingSnapshot, BS1, merged together. If both the primary and secondary BuildingSnapshot have data for a given field, the primary’s fields are preferred and merged into the child, B3.

The last sentence should say something like

* "the secondary's fields are preferred" since Stephen made that change that the latest data uploaded becomes what you see in the canonical building.
I don't really like that terminology anyway, but we can do a broader pass on the docs sometime in the future.
(what does "primary" or "secondary" mean, and what determines if it is primary or secondary?)

**Comment by StephenCzarnecki** [comment]
While looking through the database and models I've found some fields that do not seem to be actually used in the app.  Recording them here for reference and to see if anyone else knows what they might be used for.

So far I have found three fields in the BuildingSnapshot table that seem to relate to canonical buildings but as best I can tell are not actually used in the code:

best_guess_canonical_building, best_guess_confidence and canonical_for_ds

best_guess_canonical_building, best_guess_confidence don't seem to ever be populated and only seem to only exist in the model and in lists of things to exclude or ignore.

canonical_for_ds is the same as best_guess_canonical_building in that it doesn't seem to be used anywhere except as something to ignore.  Unlike the first two there is a comment in the code saying "None if Snapshot is not canonical, otherwise points to Dataset pk."  However this does not seem to be true in practice.  If I upload a file and go through mapping and matching there is a canonical record but the canonical_for_ds field is still empty.

**Comment by StephenCzarnecki** [comment]
Made a branch with some initial notes here:  https://github.com/SEED-platform/seed/tree/645_update_data_model_docs  I want to at least clean up formatting a bit and add a small section about what happens to the CanonicalBuilding table on importing a new record before making a pull request to incorporate.

**Comment by StephenCzarnecki** [comment]
@rebrownlbnl @RDmitchell I changed the doc a bit and added a few sections trying to explain exactly what happened to the database and when when the user imports a record.  I mostly left the existing text since it is basically correct at a high level.  There were a few things that were incorrect due to changes we made so I fixed that.  

If you have a minute to take a quick look and see if it is too detailed and/or confusing and/or has any other issues.  Screenshots may be helpful and it might be possible to include/embed them in a rst file so I can look into that if you think it would be helpful.  I primarily dealt with the BuildingSnapShot table but talk about some of the other interactions like the organizations and CanonicalBuilding tables.

Rich - I also tried to fix the "primary"/"secondary" terminology by instead referring them to "existing" and "new".  Might not be exactly accurate but I think it somewhat works.  Let me know if you think of anything better.
